### PR TITLE
Feature/tk 1 create classes

### PR DESCRIPTION
**Pull request by jordan-hanson**

### CODE REVIEW

**Review by kappamax — CHANGES_REQUESTED** [review]
Can you use .gitignore to ignore all your .gradle and .idea files, which will make your changes and PRs look a lot easier to read through?

File: src/main/java/CashDrawer.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        return total;
+    }
+
+    public void setTotal() {
+        this.total = (20 * twenty)  + (10 * ten) + (5 * five) + (2 * two) + (1 * one);
+    }
+
+    @Override
+    public String toString() {
+        return "$" + total +
```
Comment (kappamax) [inline src/main/java/CashDrawer.java:75]: Do you know the difference between "$" + total + " " + twenty and a String builder vs a String formatter?

This one deserves some looking into. Here are some great reads about this.
https://kylewbanks.com/blog/java-string-concatenation-vs-stringbuilder-vs-string-format-performance
https://www.baeldung.com/java-string-performance

**Comment by jordan-hanson** [comment]
Ya I have it, but I’m not sure why it isn’t working. I think when I tried creating a jar file I changed a configuration somewhere. Any ideas??

File: src/main/java/CashDrawer.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
+public class CashDrawer {
+    // Include Cash values to build a CashDrawer Array of a Cash Register
+    int twenty;
```
Comment (kappamax) [inline src/main/java/CashDrawer.java:3]: Should these have private of public or package access modifiers? Rec you make them private.

File: src/main/java/CashDrawer.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    int two;
+    int one;
+    int total;
+
+    //Build out two constructors Include all properties, and a Default
+
+    public CashDrawer() {
+    }
+
+    public CashDrawer(int twenty, int ten, int five, int two, int one) {
```
Comment (kappamax) [inline src/main/java/CashDrawer.java:15]: This constructor is interesting, with the implementation, what happens if you pass in negative numbers into the args? Can the register end up in an invalid state?

File: src/main/java/CashDrawer.java (hunk 1 above)
Comment (KrbAlmryde) [reply src/main/java/CashDrawer.java:75]: 2nd the StringBuilder or String Formatter approach. From a performance standpoint its pretty wild just how inefficient string concatenation is. Also, pro-tip, this is a common java interview question (or was) so good to have in your back pocket ;)

**Comment by KrbAlmryde** [comment]
> Ya I have it, but I’m not sure why it isn’t working. I think when I tried creating a jar file I changed a configuration somewhere. Any ideas??

The problem is you havent committed the `.gitignore` file yet, so git doesnt know what to exclude. Try adding the following to a .gitignore file, add and committing that file and see what happens


```
HELP.md
.gradle
build/
!gradle/wrapper/gradle-wrapper.jar
!**/src/main/**/build/
!**/src/test/**/build/

### STS ###
.apt_generated
.classpath
.factorypath
.project
.settings
.springBeans
.sts4-cache
bin/
!**/src/main/**/bin/
!**/src/test/**/bin/

### IntelliJ IDEA ###
.idea
*.iws
*.iml
*.ipr
out/
!**/src/main/**/out/
!**/src/test/**/out/

### NetBeans ###
/nbproject/private/
/nbbuild/
/dist/
/nbdist/
/.nb-gradle/

### VS Code ###
.vscode/
```

**Pull request merged**